appjs/views/edit_project.js

```diff
@@ -404,16 +483,19 @@ var EditProject = BaseView.extend(require('./mixins/actions'), require('./mixins
         if ( view.model.hasType( 'graphic' ) || view.model.hasPreviewType('live') ) {
           // Setup our iframe with pym
           if ( view.pym ) { view.pym.remove(); }
-          view.pym = new pym.Parent('embed-preview', previewUrl);
-          view.pym.iframe.onload = iframeLoaded;
-
+          if ( view.formValidate(view.model, view.$('#projectForm')) ){
+            view.pym = new pym.Parent('embed-preview', previewUrl);
+            view.pym.iframe.onload = iframeLoaded;
+          }
```

Comment:
Only load the preview if the form is valid. Prevents the project preview from loading until all/enough information has been entered in order to generate a correct preview.